tests/parameter_handler/parameter_acceptor_06.output

```diff
@@ -18,7 +18,7 @@ DEAL::\label{parameters:First_20Class}
 DEAL::
 DEAL::\begin{itemize}
 DEAL::\item {\it Parameter name:} {\tt First bool}
-DEAL::\phantomsection\label{parameters:First Class/First bool}
+DEAL::\phantomsection\label{parameters:First_20Class/First_20bool}
```

Comment:
I think that this is causing us unnecessary pain in ASPECT. What if you left the labels as they are, but escaped as necessary -- i.e., where you current change `demangle(section_name)` to `escape(section_name)`, instead use `escape(demangle(section_name))`? This shouldn't actually affect much of anything in ASPECT but still has the desired effect -- I learned that almost anything can appear inside a label tag.

Reply:
> I think that this is causing us unnecessary pain in ASPECT.

I am not happy with the current logic because it is not consistent (see my comment on the PR with the examples). It doesn't make sense to keep it that way.

> What if you left the labels as they are, but escaped as necessary?

No, that doesn't work. Things like ``\%`` or ``\\`` can not be inside labels not escaping them and leaving them as ``%`` and ``\`` also fails. The only reasonable way I can think of is to mangle everything except spaces and dashes but even then we wouldn't be backwards compatible.

Reply:
In other words, one could say that you're fixing a bug :-)

Reply:
> That's what I meant: you only escape what is necessary inside labels. [...] But it's backward compatible for all cases where the current behavior works with latex.

I don't understand. We are currently escaping spaces in section labels, see https://github.com/dealii/dealii/pull/5385#issuecomment-341797701 so how would this be backwards compatible?

Reply:
Ah, I see. You could of course treat section names differently than parameter names, but that's annoying to do things differently. We could output the old and the new way for the labels so that `\ref` statements continue to work, but that too does not sound like an appealing solution.

Reply:
> but that too does not sound like an appealing solution.

I wouldn't mind doing that. This would make the transition for us in ASPECT much easier.

Reply:
Then go with it. It's backward compatible, it's easy, and reduces pain downstream, and it's likely only a few extra lines of code.